ckcp/gitops.sh

```diff
@@ -86,33 +86,37 @@ check_cluster_role(){
   fi
 }
 
+install_app(){
+  APP="$1"
 
+  echo -n "  - $APP application: "
+  oc apply -f "$GITOPS_DIR/$APP.yaml" --wait >/dev/null
+  argocd app wait "$APP" >/dev/null
+  echo "OK"
+}
 
 install_openshift_gitops(){
   APP="openshift-gitops"
-  export KUBECONFIG="$KUBECONFIG_CLUSTER"
 
   local ns="openshift-operators"
 
   #############################################################################
   # Install the gitops operator
   #############################################################################
   echo -n "  - OpenShift-GitOps: "
-  if ! oc get subscriptions -n "$ns" openshift-gitops-operator >/dev/null 2>&1; then
-    oc apply -f "$GITOPS_DIR/$APP.yaml" --wait >/dev/null
-  fi
+  oc apply -f "$GITOPS_DIR/$APP.yaml" --wait >/dev/null
   echo "OK"
 
   #############################################################################
   # Wait for the URL to be available
   #############################################################################
   echo -n "  - ArgoCD dashboard: "
   test_cmd="oc get route/openshift-gitops-server --ignore-not-found -n $APP -o jsonpath={.spec.host}"
-  ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  ARGOCD_HOSTNAME="$($test_cmd})"
```

Comment:
should be "$(${test_cmd})"

Reply:
I guess both will work, but for being on the safer side I would suggest using the current one, bash array make sure that spaces are preserved. Please correct me if this is wrong.

Refer: https://mywiki.wooledge.org/BashFAQ/050 5th example

Reply:
Test command itself is not stored as array, it's just one element:
```
$ test_cmd="oc get route/openshift-gitops-server --ignore-not-found -n $APP -o jsonpath={.spec.host}"
$ echo ${test_cmd[0]}
oc get route/openshift-gitops-server --ignore-not-found -n -o jsonpath={.spec.host}
$ echo ${test_cmd[1]}

```

Reply:
I usually try to avoid arrays since it's not clear to anyone what it's doing. In this case we do not expect arguments with space inside.

Reply:
Ok now this is clear to me, thanks :)